inc/bdx86_core.h

```diff
@@ -347,15 +347,13 @@ typedef ND_UINT32 ND_REG_SIZE;
 // Misc macros.
 //
 
-// Sign extend 8 bit to 64 bit.
-#define ND_SIGN_EX_8(x)             (((x) & 0x00000080) ? (0xFFFFFFFFFFFFFF00 | (x)) : ((x) & 0xFF))
-// Sign extend 16 bit to 64 bit.
-#define ND_SIGN_EX_16(x)            (((x) & 0x00008000) ? (0xFFFFFFFFFFFF0000 | (x)) : ((x) & 0xFFFF))
-// Sign extend 32 bit to 64 bit.
-#define ND_SIGN_EX_32(x)            (((x) & 0x80000000) ? (0xFFFFFFFF00000000 | (x)) : ((x) & 0xFFFFFFFF))
-// Wrapper for for ND_SIGN_EX_8/ND_SIGN_EX_16/ND_SIGN_EX_32. Sign extend sz bytes to 64 bits.
-#define ND_SIGN_EX(sz, x)           ((sz) == 1 ? ND_SIGN_EX_8(x) : (sz) == 2 ? ND_SIGN_EX_16(x) :                   \
-                                     (sz) == 4 ? ND_SIGN_EX_32(x) : (x))
+#define ND_UINT_SZ_MASK(sz)         ( ((ND_UINT64) -1) >> ( ( (8 - (sz) ) & 0x7 ) * 8) )
```

Comment:
There already exists a macro for this purpose - `ND_SIZE_TO_MASK`. I would rewrite that macro like this instead of creating a new one: 
```
#define ND_SIZE_TO_MASK(sz)         (0xFFFFFFFFFFFFFFFFull >> ((8 - (sz)) * 8))
```

Reply:
I used that macro but didn't change it yet. I'll leave that for the future.